view/next-project/src/components/sponsoractivities/SponsorActivitiesAddModal.tsx

```diff
@@ -58,26 +48,57 @@ export default function SponsorActivitiesAddModal(props: Props) {
     userID: users[0].id || 0,
     isDone: false,
     feature: 'なし',
+    design: 0,
+    url: '',
     expense: 0,
-    remark: '',
+    remark: REMARK_PAMPHLET,
   });
 
-  const [design, setDesign] = useState<string>('学生が作成');
-  useEffect(() => {
-    const newRemark =
-      design === '学生が作成'
-        ? REMARK_DESIGN_STUDENT + REMARK_PAMPHLET
-        : design === '企業が作成'
-        ? REMARK_DESIGN_SPONSOR + REMARK_PAMPHLET_SPONSOR
-        : REMARK_DESIGN_OTHER + REMARK_PAMPHLET_OTHER;
-    const newRemarkFeature =
-      formData.feature === 'ポスター'
-        ? REMARK_POSTER
-        : formData.feature === 'クーポン'
-        ? REMARK_COUPON
-        : '';
-    setFormData({ ...formData, remark: newRemark + newRemarkFeature });
-  }, [design]);
+  const setDesign =
+    (input: string) =>
+    (
+      e:
+        | React.ChangeEvent<HTMLSelectElement>
+        | React.ChangeEvent<HTMLInputElement>
+        | React.ChangeEvent<HTMLTextAreaElement>,
+    ) => {
+      const remarkOption =
+        formData.feature === 'ポスター'
+          ? REMARK_POSTER
+          : formData.feature === 'クーポン'
+          ? REMARK_COUPON
+          : '';
+      const newRemarkDesign = e.target.value === '0' ? REMARK_PAMPHLET : '';
+      console.log(formData.remark);
+      console.log(e.target.value);
+      setFormData({
+        ...formData,
+        design: Number(e.target.value),
+        remark: remarkOption + newRemarkDesign,
+      });
+    };
+
+  const setFeature =
+    (input: string) =>
+    (
+      e:
+        | React.ChangeEvent<HTMLSelectElement>
+        | React.ChangeEvent<HTMLInputElement>
+        | React.ChangeEvent<HTMLTextAreaElement>,
+    ) => {
+      const newRemarkFeature =
+        e.target.value === 'ポスター'
+          ? REMARK_POSTER
+          : e.target.value === 'クーポン'
+          ? REMARK_COUPON
+          : '';
+      const remarkDesign = formData.design === 0 ? REMARK_PAMPHLET : '';
+      setFormData({
+        ...formData,
+        [input]: e.target.value,
+        remark: newRemarkFeature + remarkDesign,
+      });
+    };
```

Comment:
特定の値を更新する場合は、inputは受け取らなくていいかな
formDataHandlerは汎用化したからそう書いてある感じかな！
```suggestion
  const setFeature = (e: React.ChangeEvent<HTMLSelectElement>) => {
    const newRemarkFeature =
      e.target.value === 'ポスター'
        ? REMARK_POSTER
        : e.target.value === 'クーポン'
        ? REMARK_COUPON
        : '';
    const remarkDesign = formData.design === 0 ? REMARK_PAMPHLET : '';
    setFormData({
      ...formData,
      feature: e.target.value,
      remark: newRemarkFeature + remarkDesign,
    });
  };
```

これを使う場合は
```
<Select value={data.feature} onChange={setFeature}>
```
こんな感じでinputを書かなくてもeventを受け取れる